### PR TITLE
SBERDOMA-51 Remove useMemo from gen.gql.react.utils

### PR DESCRIPTION
**Pull request by dkoviazin**
Instead of wrighting 
```code js
 useEffect(() => {
   // ....
  }, [objects])
```
Use loading flag
```code js
 useEffect(() => {
   // ....
  }, [loading])
```

### CODE REVIEW

**Comment by pahaz** [comment]
Could you please also fix it here: https://github.com/open-condo-software/condo/blob/master/apps/condo/domains/common/utils/codegeneration/generate.hooks.ts#L53

File: apps/condo/domains/common/utils/codegeneration/generate.hooks.ts [hunk 1]
```diff
@@ -18,29 +18,31 @@ interface IHookConverters<GQL, GQLInput, UI, UIForm> {
...
-    function useObjects (variables: Q, memoize = true) {
+    function useObjects (variables: Q) {
         const intl = useIntl()
         const ServerErrorPleaseTryAgainLaterMsg = intl.formatMessage({ id: 'ServerErrorPleaseTryAgainLater' })
         const AccessErrorMsg = intl.formatMessage({ id: 'AccessError' })
 
         // eslint-disable-next-line prefer-const
         const result = useQuery<{ objs?: GQL[], meta?: { count?: number } }, Q>(gql.GET_ALL_OBJS_WITH_COUNT_QUERY, {
             variables,
+            fetchPolicy: 'no-cache', // Without this flag table will shake on sorting
```
Comment (pahaz) [inline apps/condo/domains/common/utils/codegeneration/generate.hooks.ts:44]: It's better to add an extra argument for apollo `useQuery`. 
Example: `useObjects( {}, { fetchPolicy: 'no-cache', ... })`

Comment (dkoviazin) [reply apps/condo/domains/common/utils/codegeneration/generate.hooks.ts]: Ok. Add queryOptions to useObjects hook.

File: apps/condo/domains/common/utils/codegeneration/generate.hooks.ts [hunk 2]
```diff
@@ -18,29 +18,30 @@ interface IHookConverters<GQL, GQLInput, UI, UIForm> {
 
 interface IHookResult<UI, UIForm, Q> {
     gql: any
-    useObject: (variables: Q, memoize?: boolean) => { obj: UI, loading: boolean, error?: ApolloError | string, refetch?: Refetch<Q> }
-    useObjects: (variables: Q, memoize?: boolean) => { objs: UI[], count: number | null, loading: boolean, error?: ApolloError | string, refetch?: Refetch<Q> }
+    useObject: (variables: Q) => { obj: UI, loading: boolean, error?: ApolloError | string, refetch?: Refetch<Q> }
+    useObjects: (variables: Q, queryOptions?: Record<string, unknown>) => { objs: UI[], count: number | null, loading: boolean, error?: ApolloError | string, refetch?: Refetch<Q> }
```
Comment (pahaz) [inline apps/condo/domains/common/utils/codegeneration/generate.hooks.ts:22]: you can import type from Apollo lib

File: packages/@core.keystone/gen.gql.react.utils.js [hunk 3]
```diff
@@ -5,39 +5,25 @@ import { useIntl } from '@core/next/intl'
...
-        const { loading, refetch, objs, count, error } = useObjects(variables, memoize)
+    function useObject (variables) {
+        const { loading, refetch, objs, count, error } = useObjects(variables)
         if (count && count > 1) throw new Error('Wrong query condition! return more then one result')
         const obj = (objs.length) ? objs[0] : null
         return { loading, refetch, obj, error }
     }
 
-    function useObjects (variables = {}, memoize = true) {
+    function useObjects (variables = {}, queryOptions = { fetchPolicy: 'no-cache', notifyOnNetworkStatusChange: true }) {
```
Comment (pahaz) [inline packages/@core.keystone/gen.gql.react.utils.js:15]: please don't use no-cache by default

Comment (pahaz) [reply packages/@core.keystone/gen.gql.react.utils.js:15]: It's ok to use no-cache on some page not by default

File: apps/condo/domains/common/utils/codegeneration/generate.hooks.ts (hunk 2 above)
Comment (dkoviazin) [reply apps/condo/domains/common/utils/codegeneration/generate.hooks.ts]: I replaced queryOptions with fetchPolicy (with apollo type FetchPolicy )

File: packages/@core.keystone/gen.gql.react.utils.js (hunk 3 above)
Comment (dkoviazin) [reply packages/@core.keystone/gen.gql.react.utils.js]: notifyOnNetworkStatusChange - is an important flag to make refetch method trigger loading indicator. 
Without using 'no-cache' table sorting will not render in table as loading indicator will not trigger 
I removed 'no-cache' from default value - but we will need to write it every time we use useObjects together with useTable

File: apps/condo/domains/common/utils/codegeneration/generate.hooks.ts [hunk 4]
```diff
@@ -18,29 +18,31 @@ interface IHookConverters<GQL, GQLInput, UI, UIForm> {
...
-        const { loading, refetch, objs, count, error } = useObjects(variables, memoize)
+    function useObject (variables: Q) {
+        const { loading, refetch, objs, count, error } = useObjects(variables)
         if (count && count > 1) throw new Error('Wrong query condition! return more then one result')
         const obj = (objs.length) ? objs[0] : null
         return { loading, refetch, obj, error }
     }
 
-    function useObjects (variables: Q, memoize = true) {
+    function useObjects (variables: Q, fetchPolicy: FetchPolicy = 'cache-first') {
```
Comment (pahaz) [inline apps/condo/domains/common/utils/codegeneration/generate.hooks.ts:36]:
```suggestion
    function useObjects (variables: Q, options?: QueryHookOptions<{ objs?: GQL[], meta?: { count?: number } }, Q>) {
```

Comment (pahaz) [reply apps/condo/domains/common/utils/codegeneration/generate.hooks.ts:36]: import { QueryHookOptions } from '@apollo/client'

File: apps/condo/domains/common/utils/codegeneration/generate.hooks.ts [hunk 5]
```diff
@@ -18,29 +18,31 @@ interface IHookConverters<GQL, GQLInput, UI, UIForm> {
...
         const intl = useIntl()
         const ServerErrorPleaseTryAgainLaterMsg = intl.formatMessage({ id: 'ServerErrorPleaseTryAgainLater' })
         const AccessErrorMsg = intl.formatMessage({ id: 'AccessError' })
 
         // eslint-disable-next-line prefer-const
         const result = useQuery<{ objs?: GQL[], meta?: { count?: number } }, Q>(gql.GET_ALL_OBJS_WITH_COUNT_QUERY, {
             variables,
+            fetchPolicy,
+            notifyOnNetworkStatusChange: true,
```
Comment (pahaz) [inline apps/condo/domains/common/utils/codegeneration/generate.hooks.ts:45]:
```suggestion
                        ...(options || {}),
```

**Review by Dimitreee — APPROVED** [review]

**Review by pahaz — APPROVED** [review]

**Pull request merged**